### PR TITLE
Make ws:// authentication more timing safe

### PR DESCRIPTION
**Pull request by Nokel81**
- Switch to crypto.timingSafeEqual() and Uint8Array (because of the
  crypto function)

Signed-off-by: Sebastian Malton <sebastian@malton.name>

follow up to #4285 

I think this is timing safe, but I don't know if the `throw new TypeError(...)`'s are.

### CODE REVIEW

File: src/main/proxy-functions/shell-api-request.ts [hunk 1]
```diff
@@ -60,9 +63,17 @@ export class ShellRequestAuthenticator extends Singleton {
     }
 
     const authToken = clusterTokens.get(tabId);
+    const buf = Uint8Array.from(token.split(",").map(ar => {
+      const res = +ar;
+
+      if (isNaN(res)) {
+        throw new TypeError("Token must be a CSV list of digits");
```
Comment (jim-docker) [inline src/main/proxy-functions/shell-api-request.ts]: could the token come in as base64 instead? Could be cleaner to handle?

Comment (Nokel81) [reply src/main/proxy-functions/shell-api-request.ts]: Probably, let me see

Comment (Nokel81) [reply src/main/proxy-functions/shell-api-request.ts]: Done

File: src/main/proxy-functions/shell-api-request.ts [hunk 2]
```diff
@@ -63,15 +63,7 @@ export class ShellRequestAuthenticator extends Singleton {
...
-    const buf = Uint8Array.from(token.split(",").map(ar => {
-      const res = +ar;
-
-      if (isNaN(res)) {
-        throw new TypeError("Token must be a CSV list of digits");
-      }
-
-      return res;
-    }));
+    const buf = Uint8Array.from(Buffer.from(token, "base64"));
```
Comment (jim-docker) [inline src/main/proxy-functions/shell-api-request.ts:66]: I guess if this fails it's not a problem. Can it throw?

**Review by jim-docker — DISMISSED** [review]

**Review by jim-docker — APPROVED** [review]

**Pull request merged**